### PR TITLE
task 5 completed

### PR DESCRIPTION
**Pull request by SimonForshaw**
Task 5 completed

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -89,24 +89,62 @@ describe("returns an review that corresponds to the review_id that is passed in"
...
-//       });
-//   });
-// });
+describe("returns all reviews from the GET reviews request", () => {
+  test("returns an array of review objects, each containing a property of review_id, title, designer, review_img_url, votes, category, owner and created_at and does not contain a property of review_body ", () => {
+    return request(app)
+      .get("/api/reviews")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.reviews).toBeSortedBy("created_at");
```
Comment (B7Bprog) [inline __tests__/app.test.js]: Please check the length of 'reviews' here.

File: model.js [hunk 2]
```diff
@@ -21,4 +21,19 @@ exports.fetchReviewsById = (inputId) => {
...
 
+exports.fetchReviews = () => {
+  return db
+    .query(
+      `SELECT reviews.review_id, reviews.title, reviews.designer, reviews.owner, reviews.review_img_url,
+  reviews.category, reviews.created_at, reviews.votes,
+  COUNT(comments.review_id) AS comment_count FROM reviews
+      LEFT JOIN comments ON reviews.review_id = comments.review_id
+      GROUP BY reviews.review_id
+      ORDER BY reviews.created_at;`
```
Comment (B7Bprog) [inline model.js]: At the end of this query string, please specify the default order to be descending.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -89,24 +89,62 @@ describe("returns an review that corresponds to the review_id that is passed in"
...
-//         expect(Array.isArray(body.reviews));
-//       });
-//   });
-// });
+describe("returns all reviews from the GET reviews request", () => {
+  test("returns an array of review objects, each containing a property of review_id, title, designer, review_img_url, votes, category, owner and created_at and does not contain a property of review_body ", () => {
+    return request(app)
+      .get("/api/reviews")
+      .expect(200)
+      .then((response) => {
```
Comment (B7Bprog) [inline __tests__/app.test.js:97]: Please also check the default descending order.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -89,24 +89,62 @@ describe("returns an review that corresponds to the review_id that is passed in"
...
+    .get("/api/reviews")
+    .then(({ body }) => {
+      expect([
+        body.reviews[0].created_at,
+        body.reviews[1].created_at,
+        body.reviews[2].created_at,
+        body.reviews[3].created_at,
+      ]).toBeSorted({ ascending: true });
+    });
+});
```
Comment (B7Bprog) [inline __tests__/app.test.js]: The following two tests don't belong to this endpoint. They should be in the describe block for /api/reviews/:review_id.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -89,24 +89,62 @@ describe("returns an review that corresponds to the review_id that is passed in"
...
+
+test("each review has the correct review_id", () => {
+  return request(app)
+    .get("/api/reviews/3")
+    .then(({ body }) => {
+      expect(body.review.review_id).toEqual(3);
+    });
+});
+test("returns with the oldest reviews first", () => {
```
Comment (B7Bprog) [inline __tests__/app.test.js:123]: This can be checked in the previous 'toBeSortedBy'.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Simon, please refactor your code according to my comments, then if all the tests pass, please merge.

**Review by elijaharley — APPROVED** [review]
Nice work on this Simon - I understand why you've commented out the console.logs in your error handlers - when you finish the project don't forget to remove these.

**Pull request merged**